docs/azure.md

```diff
@@ -5,171 +5,305 @@
-3. Create a Storage account and, within that, an Azure Blob Container in the same location where the Batch account was created. Take note of the account name and key.
-4. If you plan to use Azure Files, create an Azure File share within the same Storage account and upload your input data.
-5. Associate the Storage account with the Azure Batch account.
-6. Make sure every process in your pipeline specifies one or more Docker containers with the {ref}`process-container` directive.
-7. Make sure all of your container images are published in a Docker registry that can be accessed by your Azure Batch environment, such as [Docker Hub](https://hub.docker.com/), [Quay](https://quay.io/), or [Azure Container Registry](https://docs.microsoft.com/en-us/azure/container-registry/) .
+You can find your access keys in the Azure Portal:
+- For Storage accounts: Navigate to your Storage account → "Access keys" in the left menu
+- For Batch accounts: Navigate to your Batch account → "Keys" in the left menu
 
-A minimal Nextflow configuration for Azure Batch looks like the following snippet:
+These keys should be kept secure and rotated periodically as they provide full access to your resources. Consider using environment variables rather than hardcoding them in configuration files, especially in shared or version-controlled environments.
 
 ```groovy
-process {
-    executor = 'azurebatch'
-}
-
 azure {
     storage {
-        accountName = "<YOUR STORAGE ACCOUNT NAME>"
-        accountKey = "<YOUR STORAGE ACCOUNT KEY>"
+        accountName = '<YOUR STORAGE ACCOUNT NAME>'
+        accountKey = '<YOUR STORAGE ACCOUNT KEY>'
     }
     batch {
-        location = '<YOUR LOCATION>'
         accountName = '<YOUR BATCH ACCOUNT NAME>'
         accountKey = '<YOUR BATCH ACCOUNT KEY>'
-        autoPoolMode = true
+        location = '<YOUR LOCATION>'
     }
 }
 ```
 
-In the above example, replace the location placeholder with the name of your Azure region and the account placeholders with the values corresponding to your configuration.
+You can also use a Shared Access Token (SAS) instead of an account key:
+
+```groovy
+azure.storage.sasToken = '<YOUR SAS TOKEN>'
+```
 
 :::{tip}
-The list of Azure regions can be found by executing the following Azure CLI command:
+When creating a SAS token, ensure you enable these permissions: `Read`, `Write`, `Delete`, `List`, `Add`, `Create` on the resource types `Container` and `Object`.
+The value of `sasToken` should be stripped of the leading `?` character.
+:::
 
-```bash
-az account list-locations -o table
-```
+:::{note}
+Environment variables can be used instead of config settings:
+- `AZURE_STORAGE_ACCOUNT_NAME` - Storage account name
+- `AZURE_STORAGE_ACCOUNT_KEY` - Storage account key
+- `AZURE_BATCH_ACCOUNT_NAME` - Batch account name
+- `AZURE_BATCH_ACCOUNT_KEY` - Batch account key
+- `AZURE_STORAGE_SAS_TOKEN` - Storage SAS token
```

Comment:
```suggestion
Environment variables can be used instead of config settings:

- `AZURE_STORAGE_ACCOUNT_NAME`: Storage account name.
- `AZURE_STORAGE_ACCOUNT_KEY`: Storage account key.
- `AZURE_BATCH_ACCOUNT_NAME`: Batch account name.
- `AZURE_BATCH_ACCOUNT_KEY`: Batch account key.
- `AZURE_STORAGE_SAS_TOKEN`: Storage SAS token.
```